src/main/java/org/apache/maven/shared/filtering/FilteringUtils.java

```diff
@@ -371,6 +375,23 @@ private static void copyFilePermissions(File source, File destination) throws IO
         }
     }
 
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    private static void setAllPermissions(File file) throws IOException {
+        if (file.exists()) {
+            try {
+                Files.setPosixFilePermissions(
+                        file.toPath(),
+                        EnumSet.of(
+                                PosixFilePermission.OWNER_READ, PosixFilePermission.OWNER_WRITE,
+                                PosixFilePermission.GROUP_READ, PosixFilePermission.GROUP_WRITE,
+                                PosixFilePermission.OTHERS_READ, PosixFilePermission.OTHERS_WRITE));
```

Comment:
I think the copied file is now world readable and writable when everything completes. That might leave arbitrary world writable files sitting around in the file system, which feels risky. Can this work without world or even group permissions? And/or can the file be set back to owner only after the copy finishes?

Reply:
It's not.  The permissions from the source file are copied, see https://github.com/apache/maven-filtering/blob/ac09a5bc42e7fe812b93106537372441a8cbd6b4/src/main/java/org/apache/maven/shared/filtering/FilteringUtils.java#L333
That's why I did not bother initially with the exec flag.

Reply:
copyFilePermissions might belong in the public copyFile method above then. As is, that's not necessarily called.

Reply:
> copyFilePermissions might belong in the public copyFile method above then. As is, that's not necessarily called.

Not sure I understand your objection.  Are you implying that `private static` static methods should always be inlined ?

Reply:
nvm, the relevant part was hidden by Github's code review folding